app/code/Magento/CatalogSearch/Model/Adapter/Mysql/Aggregation/DataProvider.php

```diff
@@ -6,65 +6,58 @@
 namespace Magento\CatalogSearch\Model\Adapter\Mysql\Aggregation;
 
 use Magento\Catalog\Model\Product;
-use Magento\CatalogInventory\Model\Stock;
-use Magento\Customer\Model\Session;
+use Magento\CatalogSearch\Model\Adapter\Mysql\Aggregation\DataProvider\SelectBuilderForAttribute;
 use Magento\Eav\Model\Config;
+use Magento\Framework\App\ObjectManager;
 use Magento\Framework\App\ResourceConnection;
 use Magento\Framework\App\ScopeResolverInterface;
 use Magento\Framework\DB\Adapter\AdapterInterface;
 use Magento\Framework\DB\Ddl\Table;
 use Magento\Framework\DB\Select;
 use Magento\Framework\Search\Adapter\Mysql\Aggregation\DataProviderInterface;
 use Magento\Framework\Search\Request\BucketInterface;
-use Magento\Framework\App\ObjectManager;
 
-/**
- * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
- */
 class DataProvider implements DataProviderInterface
 {
     /**
      * @var Config
      */
     private $eavConfig;
 
-    /**
-     * @var Resource
-     */
-    private $resource;
-
     /**
      * @var ScopeResolverInterface
      */
     private $scopeResolver;
 
     /**
-     * @var Session
+     * @var AdapterInterface
      */
-    private $customerSession;
+    private $connection;
 
     /**
-     * @var AdapterInterface
+     * @var SelectBuilderForAttribute
      */
-    private $connection;
+    private $selectBuilderForAttribute;
```

Comment:
Need to lead all of the names (related to adding stock condition) to familiar clear naming

Reply:
Done.